src/components/popupWithConfirm.js

```diff
@@ -0,0 +1,19 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._handleFormSubmit = handleFormSubmit;
+    this._confirmBtn = this._popupElement.querySelector(".modal__button");
+  }
+
+  setEventListeners() {
+    //modify to work with modals without inputs
+    super.setEventListeners();
+    this._confirmBtn.addEventListener("click", (evt) => {
+      evt.preventDefault(); //prevent page refresh
+      this._handleFormSubmit(this); //call handleFormSubmit
+      this.closePopup();
```

Comment:
` this.closePopup();`


Please, delete it

You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well